var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -323,16 +323,12 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('adios2', when='+adios2')
     depends_on('blas')
-    depends_on('boost', when='+boost')
+    depends_on(Boost.with_default_variants, when='+boost')
     # Need to revisit the requirement of STK
-    depends_on('boost', when='+stk')
+    depends_on(Boost.with_default_variants, when='+stk')
```

Comment:
I know someone who works on Trilinos. Let me ask them what would be best here.

```
$ grep -hPiR "#include\s*<boost" Trilinos/ | sort -u
#include <boost/algorithm/string.hpp>
#include <boost/config.hpp>
#include <boost/exception/all.hpp>
#include <boost/functional/hash.hpp>
#include <boost/graph/adjacency_list.hpp>
#include <boost/graph/connected_components.hpp>
#include <boost/graph/distributed/adjacency_list.hpp>
#include <boost/graph/distributed/connected_components_parallel_search.hpp>
#include <boost/graph/distributed/graphviz.hpp>
#include <boost/graph/distributed/mpi_process_group.hpp>
#include <boost/graph/graphviz.hpp>
#include <boost/graph/kruskal_min_spanning_tree.hpp>
#include <boost/graph/prim_minimum_spanning_tree.hpp>
#include <boost/graph/use_mpi.hpp>
#include <boost/lexical_cast.hpp>
#include <boost/math/special_functions/ellint_rf.hpp>
#include <boost/math/special_functions/ellint_rj.hpp>
#include <boost/math/tools/toms748_solve.hpp>
#include <boost/optional/optional_io.hpp>
#include <boost/property_map/parallel/distributed_property_map.hpp>
#include <boost/property_tree/ptree.hpp>
#include <boost/property_tree/xml_parser.hpp>
#include <boost/stacktrace.hpp>
#include <boost/tuple/tuple.hpp>
  #include <boost/type_traits/is_polymorphic.hpp>
#include <boost/unordered_map.hpp>
#include <boost/version.hpp>
```

Reply:
Many of those might be in tests...

Reply:
Oh, I didn't check that. It looks like they are all in 'packages/'.

```
Trilinos/packages/minitensor/src/MiniTensor_Tensor.h:#include <boost/tuple/tuple.hpp>
Trilinos/packages/minitensor/src/MiniTensor_Matrix.h:#include <boost/tuple/tuple.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/use_mpi.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/distributed/adjacency_list.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/connected_components.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/distributed/connected_components_parallel_search.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/property_map/parallel/distributed_property_map.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/distributed/mpi_process_group.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/graph/distributed/graphviz.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/version.hpp>
Trilinos/packages/percept/src/percept/mesh/geometry/stk_geom/3D/FitGregoryPatchesPBGL.cpp:#include <boost/optional/optional_io.hpp>
Trilinos/packages/ifpack/src/supportgraph/Ifpack_SupportGraph.h:#include <boost/graph/adjacency_list.hpp>
Trilinos/packages/ifpack/src/supportgraph/Ifpack_SupportGraph.h:#include <boost/graph/kruskal_min_spanning_tree.hpp>
Trilinos/packages/ifpack/src/supportgraph/Ifpack_SupportGraph.h:#include <boost/graph/prim_minimum_spanning_tree.hpp>
Trilinos/packages/ifpack/src/supportgraph/Ifpack_SupportGraph.h:#include <boost/config.hpp>
Trilinos/packages/krino/krino/krino_lib/Akri_MathUtil.cpp:#include <boost/math/tools/toms748_solve.hpp>
Trilinos/packages/pamgen/rtcompiler/Registrar.C:#include <boost/math/special_functions/ellint_rj.hpp>
Trilinos/packages/pamgen/rtcompiler/Registrar.C:#include <boost/math/special_functions/ellint_rf.hpp>
Trilinos/packages/fei/base/fei_MapType.hpp:#include <boost/unordered_map.hpp>
Trilinos/packages/muelu/src/Utils/MueLu_BoostGraphviz.hpp:#include <boost/graph/graphviz.hpp>
Trilinos/packages/rol/src/compatibility/boost/property_tree/ROL_ParameterList.hpp:#include <boost/algorithm/string.hpp>
Trilinos/packages/rol/src/compatibility/boost/property_tree/ROL_ParameterList.hpp:#include <boost/lexical_cast.hpp>
Trilinos/packages/rol/src/compatibility/boost/property_tree/ROL_ParameterList.hpp:#include <boost/property_tree/ptree.hpp>
Trilinos/packages/rol/src/compatibility/boost/property_tree/ROL_ParameterList.hpp:#include <boost/property_tree/xml_parser.hpp>
Trilinos/packages/rol/src/compatibility/boost/stacktrace/ROL_stacktrace.hpp:#include <boost/stacktrace.hpp>
Trilinos/packages/rol/src/compatibility/boost/stacktrace/ROL_stacktrace.hpp:#include <boost/exception/all.hpp>
Trilinos/packages/teuchos/cmake/CheckForBoostIsPolymorphic.cmake:  #include <boost/type_traits/is_polymorphic.hpp>
Trilinos/packages/stokhos/src/epetra/Stokhos_AdaptivityManager.hpp:#include <boost/unordered_map.hpp>
Trilinos/packages/stokhos/src/epetra/Stokhos_AdaptivityManager.cpp:#include <boost/functional/hash.hpp>
```

Reply:
@asarkar-parsys, maybe use this to get trilinos straightened out and we can merge?

Reply:
It looks like `boost+graph+math+exception+stacktrace` should give complete coverage. I'm not sure if it should be constrained with `+stk`.

Reply:
As long as boost remains a variant of trilinos and not a requirement I'm fine with the suggested package list! If anyone needs finer granularity of the variants we can always take another PR :)